src/modules/auth_ephemeral/auth_ephemeral_mod.c

```diff
@@ -226,12 +230,22 @@ static inline int add_secret(str _secret_key)
 	memset(secret_struct, 0, sizeof (struct secret));
 	secret_struct->secret_key = _secret_key;
 	SECRET_LOCK;
-	if (secret_list != NULL)
+	if (secret_list == NULL)
+	{
+		secret_list = (struct secret **) shm_malloc(sizeof(struct secret *));
+		if (secret_list == NULL)
+		{
+			LM_ERR("unable to allocate shared memory\n");
+			return -1;
```

Comment:
You probably want to free the previously allocated secret_struct

Reply:
Fixed.